packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.props

```diff
@@ -2,8 +2,28 @@
   <PropertyGroup>
     <_MicrosoftNetSdkWindowsDesktop>true</_MicrosoftNetSdkWindowsDesktop>
   </PropertyGroup>
-  
-  <ItemGroup Condition=" '$(EnableDefaultItems)' == 'true' And '$(UseWPF)' == 'true'">
+
+  <PropertyGroup>
+    <!-- 
+    WindowsDesktop SDK supports WPF and WinForms on 
+      - .NET Core 3.0 and greater
+      - .NET Framework 3.0 and greater
+   
+    Note that on .NET Framework versions < 4.0, additional workarounds may be required to build applications 
+    using the SDK style projects. For e.g., see https://github.com/microsoft/msbuild/issues/1333
+    
+    Irrespective of whether '$(TargetFrameworkIdentifier)' is '.NETCoreApp' or '.NETFramework', 
+    the minimum value of $(_TargetFrameworkVersionWithoutV) we will be testing for is '3.0'
+    
+    Note:
+      Please see https://github.com/microsoft/msbuild/issues/3212 for a discussion about the use of 
+      the private $(_TargetFrameworkVersionWithoutV) property - which will likely remain supported and
+      is safe to use here. 
+  -->
+    <_WindowsDesktopSdkTargetFrameworkVersionFloor>3.0</_WindowsDesktopSdkTargetFrameworkVersionFloor>
```

Comment:
The experience in SDK style projects for .NET Framework 3.0, 3.5 is a bit clunky. It's certainly not great for .NET Framework 1.0, 2.0 etc. 

There is a ref-assembly nuget package (https://www.nuget.org/packages?q=Microsoft.NETFramework.ReferenceAssemblies), but `ResGen.exe` doesn't work under `dotnet.exe` when building for .NET 3.0, .NET 3.5. I couldn't find targeting pack/ref-assembly nuget for .NET Framework 1.0, 2.0 etc . 

Given this, I think it would be best to limit the capability of WindowsDesktop SDK to .NET Framework 3.0+. 

WinForms is supported from .NET Framework 1.0 onwards - so we'd be reducing a bit of the 'supported profile' (1.0, 2.0 etc. doesn't work today). This shouldn't pose a problem for WPF (which is .NET 3.5+). 


/cc @merriemcgaw, @nguerrera, @dsplaisted

Reply:
I'm totally OK with that @vatsan-madhavan. Technically 1.0 and 1.1 are out of mainstream support.